src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1223,18 +1223,21 @@ protected NodeModel(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPo
             RaisesModificationEvents = true;
         }
 
+
+        internal string cachedAsmName = null;
+
         /// <summary>
         /// The method returns the assembly name from which the node originated.
         /// </summary>
         /// <returns>Assembly Name</returns>
-        internal virtual AssemblyName GetNameOfAssemblyReferencedByNode()
+        internal virtual string GetNameOfAssemblyReferencedByNode()
```

Comment:
@dimven-adsk thanks for these changes - the cache is a good idea.
What I would prefer is that unless the change from `AssemlyName` -> `String` has clear performane benefits I would say to keep the signature of these methods the same.

I assume that the real perf gain here is from the cache and not using the GetType().Assembly  / strange double lookup / allocation of `AssemblyName`. It seems if we just cache the `AssemblyName` we'll get most of the benefit, though I guess the `AssemblyName` objects are quite wasteful at the moment though I can see us using some of these other properties in the future and I'd rather avoid spreading out the cost of allocating these later in a less predictable manner, or doing error prone string munging - thoughts?

Reply:
I would echo the same:
`What I would prefer is that unless the change from AssemlyName -> String has clear performane benefits I would say to keep the signature of these methods the same.`